client/errors.go

```diff
@@ -53,3 +55,28 @@ func newErrMissingResponseField(name string) error {
 func newErrInvalidResponseField(name string, err error) error {
 	return fmt.Errorf("invalid %s field in the response: %w", name, err)
 }
+
+type SignError struct {
+	err error
```

Comment:
Looks like you really like `errors.Join`, but can't use it here.

Reply:
Yes, it is useful, but can't use it yet